src/ivc/public_params.rs

```diff
@@ -34,12 +34,12 @@ where
     SC: StepCircuit<ARITY, C::Scalar>,
     RP: ROPair<C::Scalar>,
 {
-    pub ck: &'key CommitmentKey<C>,
-    pub params: SynthesizeStepParams<C::Scalar, RP::OnCircuit>,
+    ck: &'key CommitmentKey<C>,
+    params: SynthesizeStepParams<C::Scalar, RP::OnCircuit>,
```

Comment:
Well first of all it is important to have `ro_constant` within `pp` from the security point of view, and secondly, we create it already inside `StepCircuitExt`. I'll try to put it out, but I don't see the difference of having generics in two classes instead of one when it will be needed inside.

Reply:
`ro_constant` is inside the `PublicParam`, just outside the `SynthesizeStepParams`. And we still hash them together, so there is no security issue.

Reply:
So synthesis time we need them to create on-circuit oracles